[Bug] [dolphinscheduler] run fail when using clickhouse sql

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

`[INFO] 2024-07-05 15:02:31.775 +0800 - Sql Params are replaced sql , parameters:
[INFO] 2024-07-05 15:02:31.776 +0800 - can't find udf function resource
[INFO] 2024-07-05 15:02:31.862 +0800 - prepare statement replace sql : SELECT *
FROM xxx_ods.s_yyy_infos_dist
WHERE device_id GLOBAL IN (
    SELECT device_id
    FROM xxx_ods.s_xxx_profile_dist final
    WHERE gmt_created >= toDateTime(now() - interval 2 hour, 3, 'Asia/Shanghai')
      AND gmt_created <= toDateTime(now(), 3, 'Asia/Shanghai')
) , sql parameters : {}
[INFO] 2024-07-05 15:02:31.863 +0800 - main statement execute query, for sql: SELECT *
FROM xxx_ods.s_yyy_infos_dist
WHERE device_id GLOBAL IN (
    SELECT device_id
    FROM xxx_ods.s_xxx_profile_dist final
    WHERE gmt_created >= toDateTime(now() - interval 2 hour, 3, 'Asia/Shanghai')
      AND gmt_created <= toDateTime(now(), 3, 'Asia/Shanghai')
) 
[ERROR] 2024-07-05 15:02:32.362 +0800 - execute sql error: Java 8 date/time type `java.time.OffsetDateTime` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling
[ERROR] 2024-07-05 15:02:32.365 +0800 - sql task error
java.lang.IllegalArgumentException: Java 8 date/time type `java.time.OffsetDateTime` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling
	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:3394)
	at org.apache.dolphinscheduler.common.utils.JSONUtils.toJsonNode(JSONUtils.java:105)
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.resultProcess(SqlTask.java:252)
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.executeQuery(SqlTask.java:318)
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.executeFuncAndSql(SqlTask.java:205)
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.handle(SqlTask.java:159)
	at org.apache.dolphinscheduler.server.worker.runner.DefaultWorkerTaskExecutor.executeTask(DefaultWorkerTaskExecutor.java:54)
	at org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecutor.run(WorkerTaskExecutor.java:175)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.OffsetDateTime` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1300)
	at com.fasterxml.jackson.databind.ser.impl.UnsupportedTypeSerializer.serialize(UnsupportedTypeSerializer.java:35)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:3389)
	... 10 common frames omitted
[ERROR] 2024-07-05 15:02:32.366 +0800 - Task execute failed, due to meet an exception
org.apache.dolphinscheduler.plugin.task.api.TaskException: Execute sql task failed
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.handle(SqlTask.java:166)
	at org.apache.dolphinscheduler.server.worker.runner.DefaultWorkerTaskExecutor.executeTask(DefaultWorkerTaskExecutor.java:54)
	at org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecutor.run(WorkerTaskExecutor.java:175)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: Java 8 date/time type `java.time.OffsetDateTime` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling
	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:3394)
	at org.apache.dolphinscheduler.common.utils.JSONUtils.toJsonNode(JSONUtils.java:105)
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.resultProcess(SqlTask.java:252)
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.executeQuery(SqlTask.java:318)
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.executeFuncAndSql(SqlTask.java:205)
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.handle(SqlTask.java:159)
	... 5 common frames omitted
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.OffsetDateTime` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1300)
	at com.fasterxml.jackson.databind.ser.impl.UnsupportedTypeSerializer.serialize(UnsupportedTypeSerializer.java:35)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:3389)
	... 10 common frames omitted
[INFO] 2024-07-05 15:02:32.373 +0800 - Get appIds from worker 192.168.1.10:1234, taskLogPath: /home/foo/apache-dolphinscheduler-3.2.1-bin/standalone-server/logs/20240705/14173763226592/2/1/2.log
[INFO] 2024-07-05 15:02:32.373 +0800 - Start finding appId in /home/foo/apache-dolphinscheduler-3.2.1-bin/standalone-server/logs/20240705/14173763226592/2/1/2.log, fetch way: log 
[INFO] 2024-07-05 15:02:32.376 +0800 - The appId is empty
[INFO] 2024-07-05 15:02:32.376 +0800 - Cancel the task successfully
[INFO] 2024-07-05 15:02:32.381 +0800 - Get a exception when execute the task, will send the task status: FAILURE to master: 192.168.1.10:1234
[INFO] 2024-07-05 15:02:32.382 +0800 - FINALIZE_SESSION`

and the s_yyy_infos_dist table schema:
`
create table s_aquaman_user_device_profile_dist
(
    id            UInt64 comment 'primary key',
   ...
    gmt_created   DateTime64(3, 'Asia/Shanghai') comment 'create time',
    gmt_modified  DateTime64(3, 'Asia/Shanghai') comment 'modify time'
)
`

### What you expected to happen

Please support DateTime type for clickhouse tasks.

### How to reproduce

1. Add clickhouse source;
2. Add project and sql workflow.
3. Get the working instance immediately when running the workflow, find failure and check the log

### Anything else

version: 3.2.1
clickhouse version: 24.1.5.6

### Version

3.2.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-bom/pom.xml
+++ b/dolphinscheduler-bom/pom.xml
@@ -360,7 +360,11 @@
                 <artifactId>jackson-core</artifactId>
                 <version>${jackson.version}</version>
             </dependency>
-
+            <dependency>
+                <groupId>com.fasterxml.jackson.datatype</groupId>
+                <artifactId>jackson-datatype-jsr310</artifactId>
+                <version>${jackson.version}</version>
+            </dependency>
             <!--protostuff-->
             <dependency>
                 <groupId>io.protostuff</groupId>

--- a/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/JSONUtils.java
+++ b/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/JSONUtils.java
@@ -25,6 +25,7 @@ import static com.fasterxml.jackson.databind.SerializationFeature.FAIL_ON_EMPTY_
 import static java.nio.charset.StandardCharsets.UTF_8;
 import static org.apache.dolphinscheduler.common.constants.DateConstants.YYYY_MM_DD_HH_MM_SS;
 
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import org.apache.commons.lang3.StringUtils;
 
 import java.io.IOException;
@@ -81,6 +82,7 @@ public final class JSONUtils {
             .addModule(new SimpleModule()
                     .addSerializer(LocalDateTime.class, new LocalDateTimeSerializer())
                     .addDeserializer(LocalDateTime.class, new LocalDateTimeDeserializer()))
+            .addModule(new JavaTimeModule())
             .defaultTimeZone(TimeZone.getDefault())
             .defaultDateFormat(new SimpleDateFormat(YYYY_MM_DD_HH_MM_SS))
             .build();

--- a/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/JSONUtils.java
+++ b/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/JSONUtils.java
@@ -79,10 +79,10 @@ public final class JSONUtils {
             .configure(READ_UNKNOWN_ENUM_VALUES_AS_NULL, true)
             .configure(REQUIRE_SETTERS_FOR_GETTERS, true)
             .configure(FAIL_ON_EMPTY_BEANS, false)
+            .addModule(new JavaTimeModule())
             .addModule(new SimpleModule()
                     .addSerializer(LocalDateTime.class, new LocalDateTimeSerializer())
                     .addDeserializer(LocalDateTime.class, new LocalDateTimeDeserializer()))
-            .addModule(new JavaTimeModule())
             .defaultTimeZone(TimeZone.getDefault())
             .defaultDateFormat(new SimpleDateFormat(YYYY_MM_DD_HH_MM_SS))
             .build();

--- a/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/JSONUtils.java
+++ b/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/JSONUtils.java
@@ -25,7 +25,6 @@ import static com.fasterxml.jackson.databind.SerializationFeature.FAIL_ON_EMPTY_
 import static java.nio.charset.StandardCharsets.UTF_8;
 import static org.apache.dolphinscheduler.common.constants.DateConstants.YYYY_MM_DD_HH_MM_SS;
 
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import org.apache.commons.lang3.StringUtils;
 
 import java.io.IOException;
@@ -61,6 +60,7 @@ import com.fasterxml.jackson.databind.node.ArrayNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.fasterxml.jackson.databind.node.TextNode;
 import com.fasterxml.jackson.databind.type.CollectionType;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import com.google.common.base.Strings;
 
 /**

--- a/dolphinscheduler-dist/release-docs/LICENSE
+++ b/dolphinscheduler-dist/release-docs/LICENSE
@@ -294,7 +294,7 @@ The text of each license is also included at licenses/LICENSE-[project].txt.
     jackson-core-asl 1.9.13: https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-core-asl/1.9.13, Apache 2.0
     jackson-databind 2.13.4: https://github.com/FasterXML/jackson-databind, Apache 2.0
     jackson-datatype-jdk8 2.13.0: https://mvnrepository.com/artifact/com.fasterxml.jackson.datatype/jackson-datatype-jdk8/2.13.0, Apache 2.0
-    jackson-datatype-jsr310 2.13.0: https://mvnrepository.com/artifact/com.fasterxml.jackson.datatype/jackson-datatype-jsr310/2.13.0, Apache 2.0
+    jackson-datatype-jsr310 2.13.4: https://mvnrepository.com/artifact/com.fasterxml.jackson.datatype/jackson-datatype-jsr310/2.13.4, Apache 2.0
     jackson-mapper-asl 1.9.13: https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-mapper-asl/1.9.13, Apache 2.0
     jackson-module-parameter-names 2.13.0: https://mvnrepository.com/artifact/com.fasterxml.jackson.module/jackson-module-parameter-names/2.13.0, Apache 2.0
     javax.jdo-3.2.0-m3: https://mvnrepository.com/artifact/org.datanucleus/javax.jdo/3.2.0-m3, Apache 2.0

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/model/JSONUtilsTest.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/model/JSONUtilsTest.java
@@ -275,14 +275,16 @@ public class JSONUtilsTest {
 
     @Test
     public void toOffsetDateTimeNodeTest() {
+        TimeZone timeZone = TimeZone.getTimeZone("UTC");
+        JSONUtils.setTimeZone(timeZone);
         LocalDateTime localDateTime = LocalDateTime.of(2024, 7, 10, 15, 0, 0);
-        OffsetDateTime offsetDateTime = OffsetDateTime.of(localDateTime, ZoneOffset.ofHours(8));
+        OffsetDateTime offsetDateTime = OffsetDateTime.of(localDateTime, ZoneOffset.ofHours(0));
         Map<String, OffsetDateTime> map = new HashMap<>();
         map.put("time", offsetDateTime);
         JsonNode jsonNodes = JSONUtils.toJsonNode(map);
         String s = JSONUtils.toJsonString(jsonNodes);
 
-        String json = "{\"time\":\"2024-07-10T15:00:00+08:00\"}";
+        String json = "{\"time\":\"2024-07-10T15:00:00Z\"}";
         Assertions.assertEquals(json, s);
     }
 

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/model/JSONUtilsTest.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/model/JSONUtilsTest.java
@@ -22,6 +22,9 @@ import org.apache.dolphinscheduler.common.utils.JSONUtils;
 import org.apache.dolphinscheduler.plugin.task.api.enums.DataType;
 import org.apache.dolphinscheduler.plugin.task.api.enums.Direct;
 
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashMap;
@@ -268,6 +271,19 @@ public class JSONUtilsTest {
         Date date = JSONUtils.parseObject(json, Date.class);
         Assertions.assertEquals(DateUtils.stringToDate("2022-02-22 13:38:24"), date);
 
+    }
+
+    @Test
+    public void toOffsetDateTimeNodeTest(){
+        LocalDateTime localDateTime = LocalDateTime.of(2024, 7, 10, 15, 0, 0);
+        OffsetDateTime offsetDateTime = OffsetDateTime.of(localDateTime, ZoneOffset.ofHours(8));
+        Map<String, OffsetDateTime> map = new HashMap<>();
+        map.put("time", offsetDateTime);
+        JsonNode jsonNodes = JSONUtils.toJsonNode(map);
+        String s = JSONUtils.toJsonString(jsonNodes);
+
+        String json = "{\"time\":\"2024-07-10T15:00:00+08:00\"}";
+        Assertions.assertEquals(json, s);
     }
 
 }

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/model/JSONUtilsTest.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/model/JSONUtilsTest.java
@@ -274,7 +274,7 @@ public class JSONUtilsTest {
     }
 
     @Test
-    public void toOffsetDateTimeNodeTest(){
+    public void toOffsetDateTimeNodeTest() {
         LocalDateTime localDateTime = LocalDateTime.of(2024, 7, 10, 15, 0, 0);
         OffsetDateTime offsetDateTime = OffsetDateTime.of(localDateTime, ZoneOffset.ofHours(8));
         Map<String, OffsetDateTime> map = new HashMap<>();

--- a/tools/dependencies/known-dependencies.txt
+++ b/tools/dependencies/known-dependencies.txt
@@ -115,7 +115,7 @@ jackson-databind-2.13.4.jar
 jackson-dataformat-cbor-2.13.3.jar
 jackson-dataformat-yaml-2.13.3.jar
 jackson-datatype-jdk8-2.13.3.jar
-jackson-datatype-jsr310-2.13.3.jar
+jackson-datatype-jsr310-2.13.4.jar
 jackson-jaxrs-base-2.13.3.jar
 jackson-jaxrs-json-provider-2.13.3.jar
 jackson-mapper-asl-1.9.13.jar

